File: Data Structures and Algorithms/LeetCode/605. Can Place Flowers.cs
public class Solution {
    public bool CanPlaceFlowers(int[] flowerbed, int n) {
        
        if (n==0) return true;
        
        if(flowerbed.Length==1){
            if(n==1 & flowerbed[0]==0) return true; else return false;
        }
             if(flowerbed.Length==2){
            if(n==1 & flowerbed[0]==0 & flowerbed[1]==0) return true; else return false;
        }    

            for(int i=1;i<flowerbed.Length-1;i++){
           if(flowerbed[0]==0 && flowerbed[1]==0) 
            { flowerbed[0]=1;
                n=n-1;}
            
            if(flowerbed.Length>2) if(flowerbed[i-1]==0 & flowerbed[i]==0 & flowerbed[i+1]==0){
                flowerbed[i]=1;
                n=n-1;
            }}
        
        if(flowerbed[flowerbed.Length-1]==0 & flowerbed[flowerbed.Length-2]==0) n=n-1;
        if(n<=0) return true;
        else return false;
 
    }
}

File: Data Structures and Algorithms/LeetCode/605. Can Place Flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        for i in range(len(flowerbed)):
            if i == 0:
                prev = 0
            else:
                prev = flowerbed[i-1]
            if i == len(flowerbed)-1:
                next = 0
            else:
                next = flowerbed[i+1]
            cur = flowerbed[i]
            if prev == 0 and cur == 0 and next == 0:
                flowerbed[i] = 1
                n -= 1
            if n <= 0:
                return True
        return False
